libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -401,8 +402,8 @@ export class BotFrameworkAdapter
         } else {
             audience = z.string().parse(oAuthScopeOrlogic);
         }
-
-        const logic = LogicT.check(oAuthScopeOrlogic) ? oAuthScopeOrlogic : LogicT.parse(maybeLogic);
+        const logicParse = LogicT.safeParse(oAuthScopeOrlogic);
+        const logic = logicParse.success ? logicParse.data : LogicT.parse(maybeLogic);
```

Comment:
Same case in this other PR https://github.com/southworks/botbuilder-js/pull/385#discussion_r1385411512

Reply:
https://github.com/southworks/botbuilder-js/pull/385#discussion_r1385452293